src/UIAutomation/FunctionalTests/UiaOperationAbstractionTests.cpp

```diff
@@ -85,6 +85,44 @@ namespace UiaOperationAbstractionTests
             ElementGetNameTest(true);
         }
 
+        // Asserts that you can get the runtime id of a UiaElement.
+        void ElementGetRuntimeIdTest(const bool useRemoteOperations)
+        {
+            ModernApp app(L"Microsoft.WindowsCalculator_8wekyb3d8bbwe!App");
+            app.Activate();
+            auto calc = WaitForElementFocus(L"Display is 0");
+
+            unique_safearray expected;
+            THROW_IF_FAILED(calc->GetRuntimeId(&expected));
+
+            auto guard = InitializeUiaOperationAbstraction(useRemoteOperations);
+
+            auto scope = UiaOperationScope::StartNew();
+
+            UiaElement element = calc;
+            auto runtimeId = element.GetRuntimeId();
+            scope.BindResult(runtimeId);
+
+            scope.Resolve();
+
+            std::vector<int> actual = *runtimeId;
+            SafeArrayAccessor<int> sa(expected.get(), VT_I4);
+            for (unsigned int i = 0; i < actual.size(); ++i)
+            {
+                Assert::AreEqual(actual[i], sa[i]);
+            }
```

Comment:
Test sizes of the two collections first.

Reply:
Done.

---
In reply to: [491090021](https://github.com/microsoft/Microsoft-UI-UIAutomation/pull/17#discussion_r491090021) [](ancestors = 491090021)

Reply:
I mentioned testing if not empty, because that would guarantee at least one iteration of the loop. :) But it's OK. You retrieve the RuntimeId via the COM API that is well-tested in that regard, so it should be all good. :)

---
In reply to: [491234267](https://github.com/microsoft/Microsoft-UI-UIAutomation/pull/17#discussion_r491234267) [](ancestors = 491234267,491090021)